File: challenge/top_interview_questions/medium_collection/array_and_strings/python/group_anagrams.py
class Solution:
	def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
		counters = []
		res = []
		
		for s in strs:
			counter = [0 for i in range(0, 26)]
			for c in s:
				counter[ord(c) - ord('a')] += 1
			try:
				index = counters.index(counter)
				res[index].append(s)
			except:
				counters.append(counter)
				res.append([s])
		return res

File: challenge/2020/04/week1/group_anagrams.cpp
class Solution {
	public:
		vector<vector<string>> groupAnagrams(vector<string>& strs) {
			vector<vector<string>> res;
			unordered_map<string, vector<string>> m;

			for (auto s : strs) {
				string sorted = s;

				sort(sorted.begin(), sorted.end());
				m[sorted].push_back(s);
			}
			for (auto x : m)
				res.push_back(x.second);
			return res;
		}
};
